google-beta/resource_compute_snapshot.go

```diff
@@ -1,220 +1,536 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
 package google
 
 import (
 	"fmt"
 	"log"
+	"reflect"
+	"strconv"
 	"time"
 
+	"github.com/hashicorp/terraform/helper/customdiff"
 	"github.com/hashicorp/terraform/helper/schema"
-	"google.golang.org/api/compute/v1"
-	"google.golang.org/api/googleapi"
+	compute "google.golang.org/api/compute/v1"
 )
 
+func customDiffComputeSnapshotSnapshotEncryptionKeys(diff *schema.ResourceDiff, meta interface{}) error {
+	// oC -> Old Convenience Value
+	// nC -> New Convenience Value
+	// oN -> Old New Field Value
+	// nN -> New New Field Value
```

Comment:
Nitpick: if we have to have a guide telling us what variable names mean, couldn't we just use more descriptive variable names? `oldConvenience`, `newConvenience`, `oldField`, and `newField` or whatever would probably make this clearer.